src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -428,96 +427,111 @@ impl HummockEventHandler {
-                            (**pinned_version).clone(),
-                        )));
+                        HummockEvent::RegisterReadVersion {
+                            table_id,
+                            new_read_version_sender,
+                        } => {
+                            let pinned_version = self.pinned_version.load();
+                            let basic_read_version = Arc::new(RwLock::new(
+                                HummockReadVersion::new((**pinned_version).clone()),
+                            ));
+
+                            let instance_id = self.generate_instance_id();
+
+                            {
+                                let mut read_version_mapping_guard =
+                                    self.read_version_mapping.write();
+
+                                read_version_mapping_guard
+                                    .entry(table_id)
+                                    .or_default()
+                                    .insert(instance_id, basic_read_version.clone());
+                            }
 
-                        let instance_id = self.instance_id_generator.fetch_add(1, Ordering::SeqCst);
+                            match new_read_version_sender.send((
+                                basic_read_version,
+                                LocalInstanceGuard {
+                                    table_id,
+                                    instance_id,
+                                    event_sender: self.hummock_event_tx.clone(),
+                                },
+                            )) {
+                                Ok(_) => {}
+                                Err(_) => {
+                                    panic!("RegisterReadVersion send fail table_id {:?} instance_is {:?}", table_id, instance_id)
+                                }
+                            }
+                        }
 
-                        {
+                        HummockEvent::DestroyReadVersion {
+                            table_id,
+                            instance_id,
+                        } => {
                             let mut read_version_mapping_guard = self.read_version_mapping.write();
-
                             read_version_mapping_guard
-                                .entry(table_id)
-                                .or_default()
-                                .insert(instance_id, basic_read_version.clone());
+                                .get_mut(&table_id)
+                                .unwrap_or_else(|| {
+                                    panic!(
+                                        "DestroyHummockInstance table_id {} instance_id {} fail",
+                                        table_id, instance_id
+                                    )
+                                })
+                                .remove(&instance_id);
```

Comment:
We may also assert the value returned from remove is some?

Reply:
fix it